src/app/components/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormArray, FormBuilder, NG_VALUE_ACCESSOR, Validators } from '@angular/forms';
+
+import { checkboxInterface } from 'src/app/core/interface/checkboxInterface';
+import { requestStatus } from 'src/app/core/models/status.enum';
+
+@Component({
+  selector: 'app-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CheckboxComponent),
+      multi: true,
+    },
+  ]
+})
+
+export class CheckboxComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: checkboxInterface;
+  disabled: boolean = false;
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit() {
+    this.form.valueChanges.subscribe((value) => {
```

Comment:
we need to unsubscribe when this component will destroy

use take until strategy   https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe